src/main/java/seedu/address/logic/commands/Command.java

```diff
@@ -11,12 +11,26 @@
 
     /**
      * Executes the command and returns the result message.
+     * Implementation of command depends on executePrimitive.
      *
      * @param model   {@code Model} which the command should operate on.
      * @param history {@code CommandHistory} which the command should operate on.
      * @return feedback message of the operation result for display
      * @throws CommandException If an error occurs during command execution.
      */
-    public abstract CommandResult execute(Model model, CommandHistory history) throws CommandException;
+    public final CommandResult execute(Model model, CommandHistory history) throws CommandException {
+        model.checkOverdue();
+        return executePrimitive(model, history);
```

Comment:
Since the execution time and order of execution is important, there should be a model.checkOverdue after executePrimitive

Reply:
I took a look at all commands and if I am right there would not be a case when I have to call checkOverdue after executePrimitive.

Reply:
This is me scrutinizing the scenario where the time it took to execute primitive could take significant enough time for a task to become overdue. But this probably won't happen with our current features.